packages/core/src/app/customer/StripeGuestForm.tsx

```diff
@@ -0,0 +1,246 @@
+    initialize(options: CustomerInitializeOptions): void;
+    onShowLogin(): void;
+}
+
+const StripeGuestForm: FunctionComponent<StripeGuestFormProps & FormikProps<GuestFormValues>> = ({
+    checkoutButtons,
+    continueAsGuestButtonLabelId,
+    isLoading,
+    initialize,
+    deinitialize,
+    onChangeEmail,
+    onShowLogin,
+    onContinueAsGuest,
+    canSubscribe,
+    requiresMarketingConsent,
+    privacyPolicyUrl,
+    step,
+}) => {
+
+    const [continueAsAGuestButton, setContinueAsAGuestButton] = useState(true);
+    const [emailValue, setEmailValue] = useState('');
+    const [authentication, setAuthentication] = useState(false);
+    const [isStripeLoading, setIsStripeLoading] = useState(true);
+    const [isNewAuth, setIsNewAuth] = useState(false);
+    const handleOnClickSubmitButton = () => {
+        onContinueAsGuest({
+            email: emailValue,
+            shouldSubscribe: false,
+        });
+    };
+    const setEmailCallback = useCallback((authenticated: boolean, email: string) => {
+        onChangeEmail(email);
+        setEmailValue(email);
+        setContinueAsAGuestButton(!email);
+        setAuthentication(authenticated);
+        if(!authenticated){
+            setIsNewAuth(true);
+        }
+    }, [setContinueAsAGuestButton, onChangeEmail]);
+
+    useEffect(() => {
+        if ((!step.isComplete || isNewAuth) && emailValue && authentication) {
+            handleOnClickSubmitButton();
+        }
+    }, [emailValue, authentication, isNewAuth]);
+
+    const handleLoading = useCallback((mounted: boolean) => {
+        setIsStripeLoading(mounted);
+    }, []);
+
+
+    const getStylesFromElement = (
+        id: string,
+        properties: string[]) => {
+        const parentContainer = document.getElementById(id);
+        if (parentContainer) {
+            return getAppliedStyles(parentContainer, properties);
+        } else {
+            return undefined;
+        }
+    };
+
+    const getStripeStyles: any = useCallback( () => {
+        const containerId = 'stripe-card-component-field';
+        const formInput = getStylesFromElement(`${containerId}--input`, ['color', 'background-color', 'border-color', 'box-shadow']);
+        const formLabel = getStylesFromElement(`${containerId}--label`, ['color']);
+        const formError = getStylesFromElement(`${containerId}--error`, ['color']);
+        return formLabel && formInput && formError ? {
+            labelText: formLabel.color,
+            fieldText: formInput.color,
+            fieldPlaceholderText: formInput.color,
+            fieldErrorText: formError.color,
+            fieldBackground: formInput['background-color'],
+            fieldInnerShadow: formInput['box-shadow'],
+            fieldBorder: formInput['border-color'],
+        } : undefined;
+    }, [])
+
+    const renderCheckoutThemeStylesForStripeUPE = () => {
+        const containerId = 'stripe-card-component-field';
```

Comment:
Can possibly move this to a constant in the file above the component declaration.